pkg/auth/autorest_auth.go

```diff
@@ -23,10 +23,23 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
-	klog "k8s.io/klog/v2"
+	"k8s.io/klog/v2"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/jongio/azidext/go/azidext"
```

Comment:
Seems this whole library https://github.com/jongio/azidext/tree/master/go/azidext should be merged with the azure-sdk-for-go-extensions since they effectively serve the same purpose.

Reply:
If its not a microsoft repo, and we aren't confident we can move the code to sdk for go extensions then we shouldn't be relying on it in the first place? 

We should be conservative when adding new dependencies to our codebase.

Reply:
Its the recommended approach from the `azure-sdk-for-go` repo:
> https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/MIGRATION.md#use-azidentity-credentials-with-older-packages

Reply:
It does have a Microsoft Corporation Copyright so looks like it is Microsoft:
> // Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

Reply:
Not being under the `Azure` path was confusing me.

Reply:
Although, wait the top level License is not Microsft:
![image](https://github.com/Azure/karpenter/assets/33269602/4f43e92b-33f4-44db-84f7-a12bfa113d70)

Reply:
> DISCLAIMER: The code in this repo is not officially supported or intended for production use. The intention of this repo it to unblock customers who would like to use the Azure.Identity capabilities in the Fluent, Resource Management, and Service Bus SDKs before they have been migrated to the new SDK Azure.Core and officially support TokenCredential. We have included minimal tests in this repo, so please take it upon yourself to fully test this code to ensure it works in your environment.

They do say that, which does make me hesitate too... but I am not sure you're worse off doing it this way vs what you had